src/widget/paintable.cpp

```diff
@@ -103,60 +104,74 @@ bool Paintable::isNull() const {
 QSize Paintable::size() const {
     if (m_pPixmap) {
         return m_pPixmap->size();
-    } else if (m_pSvg) {
+    }
+
+    if (m_pSvg) {
         return m_pSvg->defaultSize();
     }
+
     return QSize();
 }
 
 int Paintable::width() const {
     if (m_pPixmap) {
         return m_pPixmap->width();
-    } else if (m_pSvg) {
+    }
+
+    if (m_pSvg) {
         QSize size = m_pSvg->defaultSize();
         return size.width();
     }
+
     return 0;
 }
 
 int Paintable::height() const {
     if (m_pPixmap) {
         return m_pPixmap->height();
-    } else if (m_pSvg) {
+    }
+
+    if (m_pSvg) {
         QSize size = m_pSvg->defaultSize();
         return size.height();
     }
+
     return 0;
 }
 
 QRectF Paintable::rect() const {
     if (m_pPixmap) {
         return m_pPixmap->rect();
-    } else if (m_pSvg) {
+    }
+
+    if (m_pSvg) {
         return QRectF(QPointF(0, 0), m_pSvg->defaultSize());
     }
+
     return QRectF();
 }
 
 QImage Paintable::toImage() const {
-    // Note: m_pPixmap is a QScopedPointer<QPixmap> and not a QPixmap.
-    // This confusion let to the wrong assumption that we could simple
-    //   return m_pPixmap->toImage();
-    // relying on QPixmap returning QImage() when it was null.
-    return !m_pPixmap ? QImage() : m_pPixmap->toImage();
+    if (m_pPixmap) {
+        return m_pPixmap->toImage();
+    }
+
+    if (m_pSvg) {
+        QImage image(m_pSvg->defaultSize(), QImage::Format_ARGB32);
```

Comment:
```suggestion
        QImage image(m_pSvg->defaultSize(), QImage::Format_ARGB32_Premultiplied);
```
![image](https://github.com/mixxxdj/mixxx/assets/12380386/4fe47653-4df6-40d0-9e30-0dca4c1027a0)